modules/SofaGeneralLoader/MeshSTLLoader.cpp

```diff
@@ -198,6 +201,9 @@ bool MeshSTLLoader::readBinarySTL(const char *filename)
 //            break;
     }
 
+    if(my_triangles.size() != (size_t)nbrFacet)
+        msg_error() << "Size mismatch between triangle vector and facetSize";
```

Comment:
should that put componentState to invalid? I don't have the global context of the method in mind.